src/hoc/RealtimeTemplates.tsx

```diff
@@ -0,0 +1,123 @@
+'use client'
+
+import { supabase } from '@/lib/supabase'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { selectCreateTemplate, setTemplates } from '@/redux/features/templateSlice'
+import store from '@/redux/store'
+import { Token } from '@/types/common'
+import { ITemplate } from '@/types/interfaces'
+import { extractImgSrcs, replaceImgSrcs } from '@/utils/signedUrlReplacer'
+import { AssigneeType, TaskTemplate } from '@prisma/client'
+import { RealtimePostgresChangesPayload } from '@supabase/supabase-js'
+import { ReactNode, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+
+interface RealTimeTemplateResponse extends ITemplate {
+  deletedAt: string
+}
+
+export const RealTimeTemplates = ({
+  children,
+  task,
+  tokenPayload,
+}: {
+  children: ReactNode
+  task?: ITemplate
+  tokenPayload: Token
+}) => {
+  const { templates } = useSelector(selectCreateTemplate)
+  const { token } = useSelector(selectTaskBoard)
+
+  const userId = tokenPayload?.internalUserId || tokenPayload?.clientId
+  const userRole = tokenPayload?.internalUserId
+    ? AssigneeType.internalUser
+    : tokenPayload?.clientId
+      ? AssigneeType.client
+      : undefined
```

Comment:
Let's make it so that a client token cannot connect to the Templates realtime channel. Otherwise they may be able to broadcast changes to templates even though the API doesn't allow it.